### ISSUE TITLE
GCR Container Registry deprecation and image source

### ISSUE THREAD

**Issue by ak185158** [GoogleCloudPlatform/prometheus-engine#505]
Will the prometheus-engine container image source be moving anytime soon with the noted deprecation of Google Container Registry?

https://cloud.google.com/container-registry/docs/deprecations/container-registry-deprecation

~~Additionally, will users be able to perform a `docker manifest inspect` at that point? Currently, we receive the below error when trying to inspect the manifest (`docker manifest inspect gke.gcr.io/prometheus-engine/prometheus:v2.35.0-gmp.5-gke.0`) in order to obtain the image sha as part of our  automation. ~~

~~`Get "https://gke.gcr.io/v2/prometheus-engine/prometheus/manifests/v2.35.0-gmp.5-gke.0": unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication`~~

~~We're able to do it with most other container image sources, however the error above seems to indicate that this is not currently allowed for this google provided image. As it stands, we're forced to manually obtain the sha for this step in our process which is not ideal.~~

Edit: We were able to find a workaround by using `docker buildx imagetools inspect --raw...` instead as it apparently does not run into this permission error. The question about if the GCR container registry deprecation will affect the container source path still stands however.

**Comment by pintohutch** [comment]
Hi @ak185158,

Thanks for inquiring! Indeed we will be moving to Artifact Registry for our hosted images and will update the location in our [public docs](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-unmanaged) when this is completed.

For now, I'll leave this ticket open for tracking in Github.

**Comment by pintohutch** [comment]
Hi @ak185158 - it looks like this migration is well under way using gcr.io [redirection](https://cloud.google.com/artifact-registry/docs/transition/auto-migrate-gcr-ar#migrate-gcrio-hosted-ar) to artifact registry (AR).

The `docker manifest inspect` should continue to work (I just tested this).

You shouldn't need to change any image URIs on your end to pull them from AR. Depending on your use case may need to [update](https://cloud.google.com/artifact-registry/docs/transition/auto-migrate-gcr-ar#required_roles) your IAM roles.

Closing for now. Let us know if you run into any issues.
